src/ledger/grain.js

```diff
@@ -185,4 +194,87 @@ export function toFloatRatio(numerator: Grain, denominator: Grain): number {
   return Number(numerator) / Number(denominator);
 }
 
+/**
+ * Splits a budget of Grain into fractional shares.
+ *
+ * It guarantees that the sum of all the splits will be exactly equal
+ * to the budget. It does not guarantee fairness, i.e. some users may
+ * receive more or less Grain than others with equal shares.
+ */
+export function splitBudget(
+  budget: Grain,
+  shares: $ReadOnlyArray<number>
+): $ReadOnlyArray<Grain> {
+  if (lt(budget, ZERO)) {
+    throw new Error("negative budget");
+  }
+  let remainingBudget = budget;
+  let remainingShares = shares.reduce((a, b) => a + b, 0);
+  if (remainingShares === 0) {
```

Comment:
Should also handle the case where remainingShares is negative. 
```suggestion
  if (remainingShares <= 0) {
```

Reply:
Sure, done.